subsys/settings/src/settings_ram.c

```diff
@@ -0,0 +1,260 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <stdbool.h>
+#include <zephyr.h>
+
+#include "settings_priv.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(settings, CONFIG_SETTINGS_LOG_LEVEL);
+
+struct settings_ram_entry {
+	sys_snode_t node;
+	char key[SETTINGS_MAX_NAME_LEN];
+	char *value; /* Dynamic allocation of values */
+	size_t value_len;
+};
+
+struct settings_read_fn_arg {
+	const char *name;
+};
+
+/* Implementations of csi_load and csi_save */
+static int settings_ram_load(struct settings_store *cs, const struct settings_load_arg *arg);
+static int settings_ram_save(struct settings_store *cs, const char *name, const char *value,
+			     size_t val_len);
+
+/* Helper function to save settings store itf */
+static struct settings_store_itf settings_ram_itf = {
+	.csi_load = settings_ram_load,
+	.csi_save = settings_ram_save,
+};
+
+/* Main struct that includes settings ram backend */
+static struct settings_store settings_ram_store = { .cs_itf = &settings_ram_itf };
+
+/* the RAM settings backend uses static SList to store settings in heap */
+static sys_slist_t settings_list = SYS_SLIST_STATIC_INIT(&settings_list);
+
+/**
+ * @brief Try to update already existing entry
+ * 
+ * Search along the settings list and try to find existing entry.
+ * If entry exist, free its value and reserve the memory for a new entry.
+ * 
+ * @param name name of the entry / key
+ * @param value value to be stored in settings
+ * @param len length of the value
+ * @return true if entry has been stored to the setting list
+ * @return false if entry does not exist or there is no RAM space to store bigger value.
+ */
+static bool try_to_update_entry(const char *name, const char *value, const size_t len)
+{
+	if (!name || len == 0 || !value) {
+		return false;
+	}
+
+	struct settings_ram_entry *entry = NULL;
+	SYS_SLIST_FOR_EACH_CONTAINER (&settings_list, entry, node) {
+		if (strncmp(entry->key, name, strnlen(entry->key, SETTINGS_MAX_NAME_LEN)) == 0) {
+			/* first free old memory */
+			k_free(entry->value);
+			entry->value = k_malloc(len);
+			if (entry->value) {
+				memcpy(entry->value, value, len);
+				entry->value_len = len;
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief try to get an entry from the settings list
+ * 
+ * Searches along the settings list and 
+ * if the entry is found update value and len arguments.
+ * 
+ * @param name name of the entry
+ * @param value pointer to memory where the entry's value will be stored
+ * @param len length of the read out entry
+ * @return true if entry has been found and read out
+ * @return false if could not find given entry
+ */
+static bool get_entry(const char *name, char *value, size_t *len)
+{
+	if (!name || !value || !len) {
+		return false;
+	}
+	struct settings_ram_entry *entry = NULL;
+	SYS_SLIST_FOR_EACH_CONTAINER (&settings_list, entry, node) {
+		if (strncmp(entry->key, name, strnlen(entry->key, SETTINGS_MAX_NAME_LEN)) == 0) {
```

Comment:
No need to use `strnlen` here (same for `try_to_update_entry` and `delete_entry`

```suggestion
		if (strncmp(entry->key, name, SETTINGS_MAX_NAME_LEN) == 0) {
```